ampligraph/latent_features/models/EmbeddingModel.py

```diff
@@ -1448,3 +1449,272 @@ def is_fitted_on(self, X):
+            if positive_base_rate is None:
+                msg = "When calibrating with randomly generated negative corruptions, " \
+                      "`positive_base_rate` must be set to a value between 0 and 1."
+                logger.error(msg)
+                raise ValueError(msg)
+
+            calibration_fn = self._calibrate_with_corruptions
+
+        dataset_handle.generate_mappings(use_all=True)
+
+        scores_pos, scores_neg = calibration_fn(dataset_handle, batch_size_pos, batch_size_neg)
+
+        scores_tf = tf.concat([scores_pos, scores_neg], axis=0)
+        labels = tf.concat([tf.ones(tf.shape(scores_pos)), tf.zeros(tf.shape(scores_neg))], axis=0)
+
+        # Platt scaling model
+        w = tf.get_variable('w', initializer=1.0, dtype=tf.float32)
+        b = tf.get_variable('b', initializer=0.0, dtype=tf.float32)
+        logits = w * tf.stop_gradient(scores_tf) + b
+
+        # Sample weights make sure the given positive_base_rate will be achieved irrespective of batch sizes
+        weigths_pos = tf.size(scores_neg) / tf.size(scores_pos)
+        weights_neg = (1.0 - positive_base_rate) / positive_base_rate
+        weights = tf.concat([tf.cast(tf.fill(tf.shape(scores_pos), weigths_pos), tf.float32),
+                             tf.cast(tf.fill(tf.shape(scores_neg), weights_neg), tf.float32)], axis=0)
+
+        loss = tf.losses.sigmoid_cross_entropy(labels, logits, weights=weights)
+
+        optimizer = tf.train.AdamOptimizer()
+        train = optimizer.minimize(loss)
+
+        epoch_iterator_with_progress = tqdm(range(1, epochs + 1), disable=(not self.verbose), unit='epoch')
+
+        with tf.Session() as sess:
+            sess.run(tf.global_variables_initializer())
+
+            for _ in epoch_iterator_with_progress:
+                losses = []
+                for batch in range(batches_count):
+                    loss_batch, _ = sess.run([loss, train])
+                    losses.append(loss_batch)
+
+                if self.verbose:
+                    msg = 'Calibration Loss: {:10f}'.format(
+                        sum(losses) / ((batch_size_pos + batch_size_neg) * batches_count))
+                    logger.debug(msg)
+                    epoch_iterator_with_progress.set_description(msg)
+
+            self.calibration_parameters = sess.run([w, b])
+
+        self.is_calibrated = True
+        dataset_handle.cleanup()
+
+    def predict_proba(self, X):
+        """
+        Predicts probabilities using the Platt scaling model (after calibration).
+
+        Model must be calibrated beforehand with the ``calibrate`` method.
+
+        Parameters
+        ----------
+        X: ndarray (shape [n, 3])
+            Numpy array of triples to be evaluated.
+
+        Returns
+        -------
+        probas: ndarray (shape [n])
+            Probability of each triple to be true according to the Platt scaling calibration.
+
+        """
+        if not self.is_calibrated:
```

Comment:
Since calibration is a must for predict_proba, I was wondering if we should make the call to calibrate api via this function(predict_proba) and have the corresponding inputs taken in as params (rather than having the user to do it separately)

Reply:
I don't think this is the right away to go, as `calibrate` is slowish and has some tunable parameters and user choices, while `predict_proba` is fast and straightforward. 

Besides, they operate on totally different datasets, so the API would be bloated, it would have to include three different datasets (X_pred_proba, X_pos, optionally X_neg) and the the other parameters from calibrate itself.